core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -949,6 +957,26 @@ public Builder setBranchSnapshot(long snapshotId, String branch) {
       return this;
     }
 
+    public Builder createSnapshotRef(String name, SnapshotRef ref) {
```

Comment:
In the builder layer would we want a separate createBranch/createTag? We do that for remove, but for creating it seems redundant. What would createBranch do differently than createTag?

Reply:
It would be fine to have this as an internal method, since it looks like this is used internally.